### ISSUE TITLE
TextYankPost and visual selection causes spurious syntax highlights

### ISSUE THREAD

**Issue by andymass** [vim/vim#4201]
**Describe the bug**

When there is a TextYankPost autocmd, visually selecting text can cause erroneous syntax highlighting to appear.  If I had to guess wildly, it could be another regex recursion issue similar to #3175, since the regex for the autocmd during visual selection might interfere with the regexes used for syntax highlighting.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Have vim with the `"*` register enabled (+clipboard, +X11) 
2. Run `valgrind --tool=memcheck --leak-check=full --show-leak-kinds=all vim --clean +'au TextYankPost * let x=10' prime.c`.

    Note: valgrind is not actually necessary, but it occurs more frequently with valgrind, possibly due to the slowdown it causes.  The file prime.c below (but any c file will work):

3. Begin visually selecting wildly
4. You should see colors occasionally jumping with wrong syntax highlighting.  Often red highlights will appear.

**Expected behavior**

There should be no spurious changes in syntax highlighting when visually selecting.

**Environment (please complete the following information):**
 - Vim version 8.1.1073
 - OS: Up to date Arch linux
 - Terminal: Konsole

**Additional context**

prime.c:
```c
#include <iostream>
using namespace std;

int main()
{
  int n, i;
  bool isPrime = true;

  cout << "Enter a positive integer: ";
  cin >> n;

  for(i = 2; i <= n / 2; ++i)
  {
      if(n % i == 0)
      {
          isPrime = false;
          break;
      }
  }
  if (isPrime)
      cout << "This is a prime number";
  else
      cout << "This is not a prime number";

  return 0;
}
```

**Comment by mg979** [comment]
This is probably related

https://github.com/mg979/vim-visual-multi/issues/43

**Comment by chrisbra** [comment]
Does adjusting the 'redrawtime' make any difference?

**Comment by andymass** [comment]
@mg979 Ah yes, that seems like the same issue, caused by the presence of the TextYankPost autocmd. 
 The animation from that thread is a good example of the behavior: https://user-images.githubusercontent.com/674812/52344790-b5e32280-2a2c-11e9-865b-ce210df0a2c0.gif .

I do use KDE's clipboard manager and when I disable it the problem seems to disappear (or become rare).

@chrisbra, no discernible difference when adjusting redrawtime (except when it's very small syntax highlighting goes away, as expected).

If I amend line_breakcheck as follows, the problem appears to go away (or perhaps become rare).
```c
    void
line_breakcheck(void)
{
    return;
}
```

**Comment by lacygoill** [comment]
Possibly fixed by [8.2.2057](https://github.com/vim/vim/commit/fccbf068f8c85474db8d8dead1530321d1f3e5b8).

**Comment by chrisbra** [comment]
closing it.
